src/Features/CSharp/Portable/InlineDeclaration/CSharpInlineDeclarationCodeFixProvider.cs

```diff
@@ -84,33 +92,66 @@ public override Task RegisterCodeFixesAsync(CodeFixContext context)
 
             if (singleDeclarator)
             {
-                // This was a local statement with a single variable in it.  Just Remove 
+                // This was a local statement with a single variable in it.  Just Remove
                 // the entire local declaration statement.  Note that comments belonging to
                 // this local statement will be moved to be above the statement containing
-                // the out-var. 
+                // the out-var.
                 var localDeclarationStatement = (LocalDeclarationStatementSyntax)declaration.Parent;
                 var block = (BlockSyntax)localDeclarationStatement.Parent;
                 var declarationIndex = block.Statements.IndexOf(localDeclarationStatement);
 
-                if (declarationIndex > 0 &&
-                    sourceText.AreOnSameLine(block.Statements[declarationIndex - 1].GetLastToken(), localDeclarationStatement.GetFirstToken()))
+                bool TryFindPriorSurvivingStatementOnSameLine(out StatementSyntax statement)
+                {
+                    statement = null;
+
+                    var localDeclarationToken = localDeclarationStatement.GetFirstToken();
+                    for (int i = declarationIndex - 1; i >= 0; i--)
+                    {
+                        var statementSyntax = block.Statements[i];
+                        if (declarationsToRemove.Contains(statementSyntax))
+                        {
+                            continue;
+                        }
+                        if (sourceText.AreOnSameLine(statementSyntax.GetLastToken(), localDeclarationToken))
+                        {
+                            statement = statementSyntax;
+                            return true;
+                        }
+                    }
+                    return false;
+                }
```

Comment:
in general, the preference is for nested functions to go at teh end of the main function, after an explicit 'return' statement of some form.

Reply:
`declarationIndex` and `localDeclarationStatement` are out of scope if i move it to the end of the method.

Reply:
Then pass them in as arguments :)   There's little need for this to be a nested function in teh first place.

Reply:
Note: it's not like this class is particularly large.  it's entire purpose is to provide the fix.  So i have no problem with this just being a helper method as well.  it's not super important that everything be scoped as narrowly as humanly possible in teh code :)